src/frequenz/sdk/microgrid/component_data.py

```diff
@@ -17,229 +19,204 @@
 from .component_states import EVChargerCableState
 
 
-class _BaseComponentData(ABC):
-    """A private base class for strongly typed component data classes."""
+@dataclass(frozen=True)
+class ComponentData(ABC):
+    """A private base class for strongly typed component data classes.
 
-    _raw: microgrid_pb.ComponentData
+    Attributes:
+        component_id: the ID identifying this component in the microgrid
+        timestamp: the timestamp of when the data was measured.
+    """
 
-    @property
-    def component_id(self) -> int:
-        """Return the component id of the message.
+    component_id: int
+    timestamp: datetime
 
-        Returns:
-            Component id of the message.
-        """
-        return self._raw.id
-
-    @property
-    def timestamp(self) -> datetime:
-        """Return the timestamp of the message.
-
-        Returns:
-            Timestamp of the message.
-        """
-        return self._raw.ts.ToDatetime(tzinfo=pytz.UTC)
-
-
-class MeterData(_BaseComponentData):
-    """A wrapper class for holding meter data."""
-
-    def __init__(self, raw: microgrid_pb.ComponentData) -> None:
-        """Create a `MeterData` instance.
+    @classmethod
+    @abstractmethod
```

Comment:
Any reason not to use `abstractclassmethod`?

Reply:
Because it is deprecated since py3.3:
https://docs.python.org/3/library/abc.html#abc.abstractclassmethod